cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -223,7 +232,14 @@ func (scaleSet *ScaleSet) GetScaleSetVms() ([]compute.VirtualMachineScaleSetVM,
 			}
 		}
 
-		allVMs = append(allVMs, vm)
+		// The resource ID is empty string, which indicates the instance may be in deleting state.
+		if len(*vm.ID) == 0 {
```

Comment:
Is there a bug for this issue, that is the resource ID empty  causing issues with the vmss?

Reply:
This is expected when the instance is in deleting state, and we won't want an empty string.